### PR TITLE
Fixed regular expressions so they deal also with CR line breaks

### PR DESCRIPTION
**Pull request by honzajavorek**
Fixing a bug I discovered when trying to employ Deckard in Apiary.

### CODE REVIEW

File: src/deckardcain.es6 [hunk 1]
```diff
@@ -1,8 +1,8 @@
 
-const API_BLUEPRINT_HEADER = /^[\uFEFF]?(((VERSION:( |\t)2)|(FORMAT:( |\t)(X-)?1A))\n)/i;
+const API_BLUEPRINT_HEADER = /^[\uFEFF]?(((VERSION:( |\t)2)|(FORMAT:( |\t)(X-)?1A))[\n\r])/i;
```
Comment (miiila) [inline src/deckardcain.es6]: Shouldn't be here (\n|\n\r)? [\n\r] would match \n or \r only IMHO.

File: src/deckardcain.es6 [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
 const API_BLUEPRINT_RESPONSE = /\+\s+Response\s+\d{3}/i;
 
-const LEGACY_BLUEPRINT_TITLE = /\-{3} ([^\n]+ )?\-{3}\n/;
+const LEGACY_BLUEPRINT_TITLE = /\-{3} ([^\n\r]+ )?\-{3}[\n\r]/;
```
Comment (miiila) [inline src/deckardcain.es6]: dtto

File: test/identify-test.es6 [hunk 3]
```diff
@@ -34,6 +34,21 @@ FORMAT: 1A
     });
   });
 
+  describe('with Windows-style line breaks and arbitrary valid content', () => {
+    const source = `\r
```
Comment (miiila) [inline test/identify-test.es6]: I think windows line break is \n\r and in this case it would be \r\n, wouldn't it?

**Comment by miiila** [comment]
Maybe I am completely wrong with my concerns, so I'd love to discuss it at least to extend my knowledge.

**Comment by honzajavorek** [comment]
You are right, thanks. I took the example from Apiary and did not think about it. It seems it's [even more complicated though](https://en.wikipedia.org/wiki/Newline). I think `[\r\n]+` should work for every case, so I'm going to go with that.

**Comment by honzajavorek** [comment]
Regular expressions in the library use new line as a boundary for certain stuff and the test is trying to ensure that even `\r` would work as the boundary. So even though `\n\r` is the right order for Windows-style new lines, I'm going to keep the test intact and I will just rename it to...

```
'with \\r line breaks and arbitrary valid content'
```

**Comment by miiila** [comment]
Slightly disagree :-) I'd vote for `[\r\n]{1,2}` if you want to have it for everything. Btw: I have omitted ZX spectrum and Acorn BBC in my previous suggestion on purpose ;-)

**Comment by honzajavorek** [comment]
Updated.

**Comment by honzajavorek** [comment]
Thanks!

**Pull request merged**